superset-frontend/plugins/legacy-plugin-chart-map-box/src/MapBox.jsx

```diff
@@ -69,87 +95,82 @@ class MapBox extends React.Component {
     }).fitBounds(bounds);
     const { latitude, longitude, zoom } = mercator;
 
-    this.state = {
-      viewport: {
+    if (initialViewportSettings === 'Fixed') {
+      setViewport({
+        longitude: !isEmpty(viewportLongitude) ? viewportLongitude : longitude,
+        latitude: !isEmpty(viewportLatitude) ? viewportLatitude : latitude,
+        zoom: !isEmpty(viewportZoom) ? viewportZoom : zoom,
+      });
+    } else {
+      setViewport({
         longitude,
         latitude,
         zoom,
-      },
-    };
-    this.handleViewportChange = this.handleViewportChange.bind(this);
-  }
+      });
+    }
+  }, []);
 
-  handleViewportChange(viewport) {
-    this.setState({ viewport });
-    const { onViewportChange } = this.props;
-    onViewportChange(viewport);
+  useEffect(() => {
+    if (viewport) {
+      // Compute the clusters based on the original bounds and current zoom level. Note when zoom/pan
+      // to an area outside of the original bounds, no additional queries are made to the backend to
+      // retrieve additional data.
+      // add this variable to widen the visible area
+      const offsetHorizontal = (width * 0.5) / 100;
+      const offsetVertical = (height * 0.5) / 100;
+      const bbox = [
+        bounds[0][0] - offsetHorizontal,
+        bounds[0][1] - offsetVertical,
+        bounds[1][0] + offsetHorizontal,
+        bounds[1][1] + offsetVertical,
+      ];
+      setClusters(clusterer.getClusters(bbox, Math.round(viewport.zoom)));
+    }
+  }, [clusterer, viewport]);
+
+  if (!viewport || !clusters) {
+    return <></>;
   }
 
-  render() {
-    const {
-      width,
-      height,
-      aggregatorName,
-      clusterer,
-      globalOpacity,
-      mapStyle,
-      mapboxApiKey,
-      pointRadius,
-      pointRadiusUnit,
-      renderWhileDragging,
-      rgb,
-      hasCustomMetric,
-      bounds,
-    } = this.props;
-    const { viewport } = this.state;
-    const isDragging =
-      viewport.isDragging === undefined ? false : viewport.isDragging;
-
-    // Compute the clusters based on the original bounds and current zoom level. Note when zoom/pan
-    // to an area outside of the original bounds, no additional queries are made to the backend to
-    // retrieve additional data.
-    // add this variable to widen the visible area
-    const offsetHorizontal = (width * 0.5) / 100;
-    const offsetVertical = (height * 0.5) / 100;
-    const bbox = [
-      bounds[0][0] - offsetHorizontal,
-      bounds[0][1] - offsetVertical,
-      bounds[1][0] + offsetHorizontal,
-      bounds[1][1] + offsetVertical,
-    ];
-    const clusters = clusterer.getClusters(bbox, Math.round(viewport.zoom));
-
-    return (
-      <MapGL
+  const isDragging =
```

Comment:
It's equivalent to `Boolean()`
```suggestion
  const isDragging = Boolean(viewport.isDragging);
```